src/pages/on-chain-governance/index.page.tsx

```diff
@@ -98,245 +84,438 @@ export default function OnChainGovernancePage(props) {
+            query: {
+              proposalType: userQueryProposalType,
+              proposalStatus: "close",
+            },
+          }}
+        >
+          <a
+            data-testid="OnChainGovernance.ClosedProposalsButton"
+            className={classNames(
+              "md:font-normal font-medium border border-l-0 rounded-r py-[6px] md:px-[25px] px-3 md:text-base text-xs text-gray-900",
+              {
+                "border-0 bg-primary-500 text-white": !isOpenProposalsClicked,
+              }
+            )}
+          >
+            <div>{windowSize <= 640 ? "Closed" : "Closed proposals"}</div>
+          </a>
+        </Link>
       </div>
-    </Container>
+    </div>
   );
 }
 
 export async function getServerSideProps(context: GetServerSidePropsContext) {
   const rpc = getWhaleRpcClient(context);
-  const response = await rpc.governance.listGovProposals({
+
+  let userQueryProposalType = ListProposalsType.ALL;
+  const userQueryProposalStatus = context.query.proposalStatus ?? "open";
+
+  switch (context.query.proposalType) {
+    case ListProposalsType.CFP:
+      userQueryProposalType = ListProposalsType.CFP;
+      break;
+    case ListProposalsType.VOC:
+      userQueryProposalType = ListProposalsType.VOC;
+      break;
+    case ListProposalsType.ALL:
+    default:
+      userQueryProposalType = ListProposalsType.ALL;
+  }
+
+  const currentBlockCount = await rpc.blockchain.getBlockCount();
+  const currentBlockInfo = await rpc.blockchain.getBlockStats(
+    currentBlockCount
+  );
+  const currentBlockMedianTime = currentBlockInfo.mediantime;
+  const allProposals = await rpc.governance.listGovProposals({
     type: ListProposalsType.ALL,
-    status: ListProposalsStatus.VOTING,
+    status: ListProposalsStatus.ALL,
   });
+
+  const queryProposals = await rpc.governance
+    .listGovProposals({
+      type: userQueryProposalType,
+      status: ListProposalsStatus.ALL,
+    })
+    .then((proposals) => {
+      proposals = proposals.filter((proposal) => {
+        if (userQueryProposalStatus === "open") {
+          return proposal.status === ProposalStatus.VOTING;
+        } else {
+          return proposal.status !== ProposalStatus.VOTING;
+        }
+      });
+      return proposals;
+    })
+    .catch((error) => {
+      console.error(error);
+      return [];
+    });
+
   return {
-    props: getOCGData(JSON.parse(JSON.stringify(response))),
+    props: getOCGData(
+      JSON.parse(JSON.stringify(allProposals)),
+      JSON.parse(JSON.stringify(queryProposals)),
+      currentBlockCount,
+      currentBlockMedianTime,
+      userQueryProposalType,
+      userQueryProposalStatus
+    ),
   };
 }
 
-function getOCGData(items: ProposalInfo[]): OCGProps {
+function getOCGData(
+  allProposals: ProposalInfo[],
+  queryProposals: ProposalInfo[],
+  currentBlockCount: number,
+  currentBlockMedianTime: number,
+  userQueryProposalType: ListProposalsType,
+  userQueryProposalStatus: string | string[]
```

Comment:
let's replace with a custom enum
```suggestion
  userQueryProposalStatus: customEnum
```